### PR TITLE
PR for product listing page

### PR DESCRIPTION
**Pull request by narharikale**
- converted all HTML to JSX
- added product listing
- added filter functionality

note: not responsive yet

### CODE REVIEW

File: src/pages/productListing/productListing.jsx [hunk 1]
```diff
@@ -1,9 +1,46 @@
...
 function ProductListing() {
-    return ( 
+
+    const { data } = useFilter();
+    
+    // const sortData =  sortedData(state , products);
+    // const filterPriceData = FilterPriceData(state , sortData);
+    // const filterCategoryData = FilterCategoryData(state , filterPriceData)
+    // const filterRatingData = FilterRatingData(state , filterCategoryData)
+
```
Comment (shraddha15-creator) [inline src/pages/productListing/productListing.jsx]: If it is not needed we can delete this?

File: src/pages/productListing/productListing.jsx [hunk 2]
```diff
@@ -1,9 +1,46 @@
...
+
+    
+
```
Comment (shraddha15-creator) [inline src/pages/productListing/productListing.jsx]: also remove empty lines

**Review by shraddha15-creator — APPROVED** [review]
Just remove the empty lines. Rest code looks good.

File: src/context/filter-context.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import { createContext , useReducer , useContext} from "react";
+import { useProducts } from "./product-context";
+import { sortedData, FilterPriceData ,FilterCategoryData , FilterRatingData , ComposeAll} from '../utils/filterutils';
```
Comment (rahulmistry751) [inline src/context/filter-context.js:3]: Suggestion: barrel export can be use here

**Review by rahulmistry751 — APPROVED** [review]

File: src/pages/productListing/productListing.jsx (hunk 1 above)
Comment (narharikale) [reply src/pages/productListing/productListing.jsx]: ohh thanks , fixed

File: src/pages/productListing/productListing.jsx (hunk 2 above)
Comment (narharikale) [reply src/pages/productListing/productListing.jsx]: fixed

File: src/context/filter-context.js (hunk 3 above)
Comment (narharikale) [reply src/context/filter-context.js:3]: as of now, this fine for me will update in future

**Pull request merged**